Neo4j.Driver/Neo4j.Driver.Tests/GqlCompliance/GqlErrorTests.cs

```diff
@@ -0,0 +1,129 @@
+﻿// Copyright (c) "Neo4j"
+// Neo4j Sweden AB [https://neo4j.com]
+// 
+// Licensed under the Apache License, Version 2.0 (the "License").
+// You may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System.Collections.Generic;
+using FluentAssertions;
+using Neo4j.Driver.Internal.ExceptionHandling;
+using Neo4j.Driver.Internal.IO.MessageSerializers;
+using Neo4j.Driver.Internal.Messaging;
+using Xunit;
+
+namespace Neo4j.Driver.Tests.GqlCompliance;
+
+public class GqlErrorTests
+{
+    [Fact]
+    public void BuildFailureMessage_ShouldDeserializeCorrectly()
+    {
+        // Arrange
+        var values = new Dictionary<string, object>
+        {
+            { "code", "Neo.ClientError.Transaction.Terminated" },
+            { "message", "Transaction terminated" },
+            { "gql_status", "GQL_STATUS" },
+            { "description", "Status description" },
+            { "diagnostic_record", new Dictionary<string, object> { { "key", "value" } } },
+            {
+                "cause",
+                new Dictionary<string, object>
+                {
+                    { "code", "Neo.ClientError.Transaction.LockClientStopped" }, { "message", "Lock client stopped" }
+                }
+            }
+        };
+
+        int majorVersion = 4;
+
+        // Act
+        var result = FailureMessageSerializer.BuildFailureMessage(values, majorVersion);
+
+        // Assert
+        result.Code.Should().Be("Neo.ClientError.Transaction.Terminated");
+        result.Message.Should().Be("Transaction terminated");
+        result.GqlStatus.Should().Be("GQL_STATUS");
+        result.GqlStatusDescription.Should().Be("Status description");
+        result.GqlDiagnosticRecord.Should().NotBeNull();
+        result.GqlDiagnosticRecord["key"].Should().Be("value");
+        result.GqlCause.Should().NotBeNull();
+        result.GqlCause.Code.Should().Be("Neo.ClientError.Transaction.LockClientStopped");
+        result.GqlCause.Message.Should().Be("Lock client stopped");
+
+        // Assert defaults added by FillGqlDefaults
+        result.GqlDiagnosticRecord.Should().ContainKey("OPERATION");
+        result.GqlDiagnosticRecord["OPERATION"].Should().Be("");
+        result.GqlDiagnosticRecord.Should().ContainKey("OPERATION_CODE");
+        result.GqlDiagnosticRecord["OPERATION_CODE"].Should().Be("0");
+        result.GqlDiagnosticRecord.Should().ContainKey("CURRENT_SCHEMA");
+        result.GqlDiagnosticRecord["CURRENT_SCHEMA"].Should().Be("/");
+
+        result.GqlRawClassification.Should().Be("");
```

Comment:
It looks like this is causing the test to fail - the actual value is `null` not an empty string.